### PR TITLE
Merge kube Accessor and client

### PR DESCRIPTION
**Pull request by nmittler**
This is a pretty large change with a lot of cleanup.

Overview of the changes:

- Moved many of the Accessor methods to utilities.
- Separated out the function of writing yaml content to files from the kube.Client API. The suite/test contexts now handle that work on behalf of the clusters. The ConfigManager interface is no longer implemented by Cluster.
- Simplified the kube.clientFactory so that it just uses an in-memory discovery client.
- Removed the kube.Deployment. It was only used in 2 places and didn't add much value.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -369,15 +407,186 @@ func (c *client) NewPortForwarder(podName, ns, localAddress string, localPort in
...
-func (c *client) PodsForSelector(ctx context.Context, namespace, labelSelector string) (*v1.PodList, error) {
-	podGet := c.Get().Resource("pods").Namespace(namespace).Param("labelSelector", labelSelector)
-	obj, err := podGet.Do(ctx).Get()
+func (c *client) PodsForSelector(ctx context.Context, namespace string, labelSelectors ...string) (*kubeApiCore.PodList, error) {
+	return c.CoreV1().Pods(namespace).List(ctx, kubeApiMeta.ListOptions{
+		LabelSelector: strings.Join(labelSelectors, ","),
+	})
+}
+
+func (c *client) ApplyYAMLFiles(namespace string, dryRun bool, yamlFiles ...string) error {
```
Comment (howardjohn) [inline pkg/kube/client.go:416]: Nit: I think I would prefer dry run to be its own function so we don't need to pollute the call with `false` everywhere? Dry run is pretty rare anyways

File: pkg/kube/client.go [hunk 2]
```diff
@@ -369,15 +407,186 @@ func (c *client) NewPortForwarder(podName, ns, localAddress string, localPort in
...
+	for _, f := range removeEmptyFiles(yamlFiles) {
+		if err := c.applyYAMLFile(namespace, dryRun, f); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (c *client) applyYAMLFile(namespace string, dryRun bool, file string) error {
+	dynamicClient, err := c.clientFactory.DynamicClient()
```
Comment (howardjohn) [inline pkg/kube/client.go:425]: should just be c.Dynamic() I think?

File: tests/integration/galley/validation_test.go [hunk 3]
```diff
@@ -86,25 +86,23 @@ func TestValidation(t *testing.T) {
...
 					}
 
-					ns := namespace.NewOrFail(t, fctx, namespace.Config{
+					ns := namespace.NewOrFail(t, ctx, namespace.Config{
 						Prefix: "validation",
 					})
 
-					_, err = cluster.ApplyContentsDryRun(ns.Name(), ym)
+					applyFiles := ctx.WriteYAMLOrFail(ctx, ym, "apply")
+					err = cluster.ApplyYAMLFiles(ns.Name(), true, applyFiles...)
```
Comment (howardjohn) [inline tests/integration/galley/validation_test.go:105]: why write to a file? applying a string directly is a better ux to me. Does the cluster API still allow this - looks like yes?

File: pkg/kube/client.go (hunk 1 above)
Comment (nmittler) [reply pkg/kube/client.go:416]: done.

File: tests/integration/galley/validation_test.go (hunk 3 above)
Comment (nmittler) [reply tests/integration/galley/validation_test.go]: This one does a dry run, which I believe is the only place in the code that uses dry run today. The ConfigManager API doesn't support dry run, so writing to files first is necessary here. I could work dry run into the ConfigManager API but I'm not sure it's worth it.

**Pull request merged**